pkg/registry/generic/rest/streamer.go

```diff
@@ -56,6 +59,16 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	if err != nil {
 		return
 	}
+
+	if resp.StatusCode == http.StatusInternalServerError {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
I think you'll need to speculatively look at the body using a LimitReader.  Also, you need to investigate other error types.  We probably need to be calling into the generic server error handling code that we use in the client so we don't do this two different ways.  That is API/errors#NewGenericServerResponse